lib/asciidoctor/extensions.rb

```diff
@@ -173,13 +173,105 @@ def process *args, &block
         raise ::NotImplementedError
       end
     end
-    #alias :process_with :process
 
     def process_block_given?
       defined? @process_block
     end
   end
 
+  module SyntaxDsl
+    include ProcessorDsl
+
+    def named value
+      # NOTE due to how processors get initialized, we must defer this assignment in some scenarios
+      if Processor === self
+        @name = value
+      else
+        option :name, value
+      end
+    end
+    # NOTE match_name may get deprecated
+    alias :match_name :named
+
+    def content_model value
+      option :content_model, value
+    end
+    alias :parse_content_as :content_model
+    alias :parses_content_as :content_model
+    #alias :parse_as :content_model
+    #alias :parsed_as :content_model
+
+    def positional_attrs *value
+      option :pos_attrs, value.flatten
+    end
+    alias :name_attributes :positional_attrs
+    alias :name_positional_attributes :positional_attrs
+
+    def default_attrs value
+      option :default_attrs, value
+    end
+
+    def resolves_attributes *args
+      args = args.fetch 0, true unless args.size > 1
+      case args
+      when true
+        option :pos_attrs, []
+        option :default_attrs, {}
+      when ::Array
+        names, defaults = [], {}
+        args.each do |arg|
+          if arg.include? '='
+            name, value = arg.split '=', 2
+            if name.include? ':'
+              idx, name = name.split ':', 2
+              idx = idx == '@' ? names.size : idx.to_i
+              names[idx] = name
+            end
+            defaults[name] = value
+          elsif arg.include? ':'
+            idx, name = arg.split ':', 2
+            idx = idx == '@' ? names.size : idx.to_i
+            names[idx] = name
+          else
+            names << arg
+          end
+        end
+        option :pos_attrs, names.compact
+        option :default_attrs, defaults
+      when ::String # QUESTION should we accept Symbol too?
```

Comment:
I found a more concise way of writing this so that we can support symbols. In other words, the two combinations are now recognized:

```ruby
resolve_attributes :precision
```

```ruby
resolve_attributes [:precision]
```

Although this is not the recommended syntax, at least the processor doesn't crash if you write it this way.